Add itolparser

boasvdp opened the pull request:
Describe your pull request here

This pull request adds itolparser, a tool that writes annotation files for iTOL (https://itol.embl.de/) from a user-supplied table. The conda package was created from the pypi package.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

boasvdp commented:
@BiocondaBot please add label

simonbray commented on recipes/itolparser/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+
+requirements:
+  host:
+    - colorbrewer
```
It should work fine with only python and pip under host

simonbray commented on recipes/itolparser/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+
+test:
+  imports:
+    - bin
```
Not really related to the conda recipe, but itolparser would be a better name for this import, could you rename the directory in your repo?

You can also create an entry point for your itolparser script: https://packaging.python.org/guides/distributing-packages-using-setuptools/#entry-points

simonbray commented on recipes/itolparser/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+  home: "https://github.com/boasvdp/itolparser"
+  license: MIT
+  license_family: MIT
+  license_file: 
```
```suggestion
  license_file: LICENSE
```

simonbray commented on recipes/itolparser/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+  license: MIT
+  license_family: MIT
+  license_file: 
+  summary: "A"
```
```suggestion
  summary: "Small script to produce iTOL colorstrip metadata files from a table"
```

simonbray commented on recipes/itolparser/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+  license_family: MIT
+  license_file: 
+  summary: "A"
+  doc_url: 
+  dev_url: 
```
```suggestion

```

boasvdp commented:
Thanks @simonbray, I'm learning a lot from your comments (first time Bioconda contributor). I have included all your suggestions in the new version. I have also updated the README and added a version flag.

simonbray commented:
@bioconda-bot please fetch artifacts

simonbray commented:
@bioconda-bot please merge

Thanks @boasvdp, this looks good to me.
